pkg/sql/sem/builtins/builtins.go

```diff
@@ -609,6 +610,65 @@ var builtins = map[string]builtinDefinition{
 		},
 	),
 
+	"gen_random_ulid": makeBuiltin(
+		tree.FunctionProperties{
+			Category: categoryIDGeneration,
+		},
+		tree.Overload{
+			Types:      tree.ArgTypes{},
+			ReturnType: tree.FixedReturnType(types.Uuid),
+			Fn: func(_ *tree.EvalContext, _ tree.Datums) (tree.Datum, error) {
+				entropy := ulid.Monotonic(rand.New(rand.NewSource(timeutil.Now().UnixNano())), 0)
```

Comment:
This creates a new random source from the current time every call to `gen_random_ulid`. This is not an effective source of entropy. It should either use a cryptographic source of randomness (as we do for UUIDs) or persist a PRNG somewhere for repeated use. (We should strongly favor the use of cryptographic randomness).

Reply:
Gotcha, I'm not too familiar with RNG but in this case, instead of using the current time, would using `crypto/rand` to also generate the seed make this acceptable?

https://golang.org/pkg/crypto/rand/ `Package rand implements a cryptographically secure random number generator.`

Specifically using https://golang.org/pkg/crypto/rand/#Int to generate the seed for the source?

Reply:
Using a cryptographic RNG to seed a PRNG is better than just seeding a PRNG with the current time, but in general it's better to just use the cryptographic RNG directly (this is what UUIDs do). Can't you just pass `crypto/rand.Reader` to `ulid.Monotonic`?

Reply:
@bdarnell Let's see if I understood you correctly.

https://github.com/cockroachdb/cockroach/pull/65768

Thank you for the review.

Reply:
Also thank you @RichardJCai for all the help!